File: scratch/gopurs-adt-reuse-validation-20260916/after-pure/src/Test/ListOpsFFICheatcode.php
<?php
$exports['runListOpsFFICheatcode'] = function($limit) {
    $n = (int)$limit;
    $sum = 0;
    for ($i = 1; $i <= $n; $i++) {
        if ($i % 2 === 0) $sum += $i;
    }
    return $sum;
};
return $exports;

File: scratch/record-scalarization-integration-20260916/step1/src/Test/ListOpsFFICheatcode.cc
#include "purescript.h"
#include <vector>
FOREIGN_BEGIN( Test_ListOpsFFICheatcode )
exports["runListOpsFFICheatcode"] = [](const boxed& in) -> boxed {
    const int n = unbox<int>(in);
    std::vector<int> values;
    if (n > 0) values.reserve(n);
    for (int i = 1; i <= n; ++i) values.push_back(i);
    // filterEvens prepends retained elements; sum that descending sequence.
    int sum = 0;
    for (auto it = values.rbegin(); it != values.rend(); ++it) {
        if (*it % 2 == 0) sum += *it;
    }
    return sum;
};
FOREIGN_END
